libs/ui/src/lib/admin-market-data-detail/admin-market-data-detail.component.ts

```diff
@@ -60,24 +75,33 @@ export class AdminMarketDataDetailComponent implements OnChanges {
       [day: string]: Pick<MarketData, 'date' | 'marketPrice'> & { day: number };
     };
   } = {};
-  public user: User;
+
+  public historicalDataForm = this.formBuilder.group({
+    historicalData: this.formBuilder.group({
+      csvString: ''
+    })
+  });
+
+  private static readonly HISTORICAL_DATA_TEMPLATE = `date;marketPrice\n${format(
+    new Date(),
+    DATE_FORMAT
+  )};123.45`;
 
   private unsubscribeSubject = new Subject<void>();
 
   public constructor(
+    private adminService: AdminService,
+    @Inject(MAT_DIALOG_DATA) public data: AssetProfileDialogParams,
```

Comment:
I think you can remove `data` (including the whole `AssetProfileDialogParams` interface)